wordpress-mobile/WordPress-iOS
# Fix deep linking in marketing emails

This PR does 2 things:

- It standardizes the naming of some static properties so that they start with a lower case character (ref: https://swift.org/documentation/api-design-guidelines/).
- It implements / fixes #11693.

## Details:

This PR implements the code to handle `/mbar` redirects, such as:

`https://public-api.wordpress.com/mbar/?redirect_to=https%3A%2F%2Fwordpress.com%2Fpost%2Fddancestore.wordpress.com`

We've also added an AASA file to `public-api.wordpress.com` (ref: pMz3w-a0N-p2).

## Dependencies:

This PR has a dependency on: https://github.com/wordpress-mobile/WordPress-iOS/pull/11851

## Matching Android PR:

https://github.com/wordpress-mobile/WordPress-Android/pull/9957

## Testing:

Make sure you erase all content and settings from your simulator before testing.  This ensures the newly installed App picks up the AASA file from public-api.wordpress.com.

### Test 1:

1. Create WP.com site.
2. Publish a post in that site.
3. Check that you received the "Congratulations on your first post!" email for the new site.
4. Tap on the blue "Start a New Post" button.
5. Make sure the editor is opened in-App.

### Test 2:

1. Launch the app.
2. Make sure you're logged in.
3. Enter this in a terminal window after replacing the blog name: `xcrun simctl openurl booted https://public-api.wordpress.com/mbar/?redirect_to=https%3A%2F%2Fwordpress.com%2Fpost%2Fdiegotest4.wordpress.com`
4. Make sure a new post is created in the App for the specified blog.

Additionally you can place breakpoints [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/024cd1353a5728e47ec199a009ba70101256e2ad/WordPress/Classes/Utility/Universal%20Links/UniversalLinkRouter.swift#L115-L117) to make sure the full URL is tracked, but the final redirect isn't.

## Update release notes:
